bionic/datatypes.py

```diff
@@ -72,15 +69,50 @@ class Result(object):
     Represents one value for one entity.
     '''
     def __init__(
-            self, query, value, file_path=None):
+            self, query, value, value_hash=None, file_path=None):
         self.query = query
         self.value = value
+        # Only present when value should persist.
         self.file_path = file_path
+        self.value_hash = value_hash
+
+    def to_hashed_result(self):
+        return HashedResult(self.query, self.value_hash)
 
     def __repr__(self):
         return f'Result({self.query!r}, {self.value!r})'
 
 
+class HashedResult(object):
+    '''
+    Represents hashed part of the result.
+    '''
+    def __init__(
```

Comment:
super nit: do we need the extra newline here?

Reply:
Nope, probably copy-pasta from above classes